tests/lib/mockMagic.py

```diff
@@ -9,3 +10,78 @@ def validate_config_for_device(self, device_mac, config):
             return return_validate_config_for_device['return_validate_intf']
         if config == validate_true['config']:
             return return_validate_config_for_device['return_validate_true']
+
+    def device_decommissioning(self, device_id, request_id):
+        """
+        mock method for cvprac device_decommissioning()
+        """
+        if device_id == device_data[0]["serialNumber"]:
+            self.result = {'value': {'key': {'requestId': request_id},
+                                     'deviceId': device_id},
+                           'time': '2022-02-12T02:58:30.765459650Z'}
+        else:
+            self.result = None
+
+    def device_decommissioning_status_get_one(self, request_id):
+        """
+        mock method for cvprac device_decommissioning_status_get_one()
+        """
+        if self.result and self.result["value"]["key"]["requestId"]:
+            resp = {"result": {"value": {"key": {"requestId": request_id},
+                                         "status": 'DECOMMISSIONING_STATUS_SUCCESS',
+                                         "statusMessage": "Disabled TerminAttr, "
+                                                          "waiting for device to be marked inactive"},
+                               "time": "2022-02-04T19:41:46.376310308Z", "type": "INITIAL"}}
+
+        else:
+            resp = {"result": {"value": {"status": 'DECOMMISSIONING_STATUS_FAILURE',
+                                         "error": "Device does not exist or is not registered to decommission"},
+                               "time": "2022-02-04T19:41:46.376310308Z", "type": "INITIAL"}}
+
+        return resp["result"]
+
+    def reset_device(self, app_name, device, create_task=True):
+        """
+        mock method for cvprac reset_device()
+        """
+        if 'parentContainerId' in device:
+            from_id = device['parentContainerId']
+        else:
+            if device['parentContainerName']:
+                from_id = device['parentContainerName']
+            else:
+                from_id = ''
+
+        if create_task:
+            if from_id and from_id == "Undefined":
+                return {'data': {'status': 'fail', 'taskIds': []}}
+            elif from_id and from_id != "Undefined":
+                return {'data': {'taskIds': ['57'], 'status': 'success'}}
+            else:
+                raise CvpApiError(msg="Error resetting device")
+        else:
+            return None
+
+        # TODO: need to write testcase - if create_task returns None, raised issue #585 in ansible-cvp
```

Comment:
@Shivani-chourasiya You can fix the reset_device in this same PR. 
https://github.com/aristanetworks/ansible-cvp/blob/f8b8e40c44dd44299b38a73409efc803281bcef9/ansible_collections/arista/cvp/plugins/module_utils/device_tools.py#L2214
Change that line to 
```
                    if resp and resp["data"]["status"] == "success":
```

Reply:
fixed